cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/MetadataStoreDataset.java

```diff
@@ -192,6 +212,23 @@ public Key build() {
         return new Key(key);
       }
     }
+
+    /**
+     * Splits {@link Key}s.
+     */
+    public static final class Splitter {
+      public static List<byte[]> split(byte[] key) {
```

Comment:
Ok, I made split() instance method of the Key class.
It is not a generic util method (because it depends on how the key parts were encoded).